tracer/tracer.go

```diff
@@ -159,6 +159,8 @@ type Config struct {
 	// IncludeEnvVars holds a list of environment variables that should be captured and reported
 	// from processes
 	IncludeEnvVars libpf.Set[string]
+	// MaxTailCalls is the number of tail calls the eBPF is allowed to make before giving up.
```

Comment:
To me making `MaxTailCalls` a user configureable option sounds counter intuitive. From a single tail call one just can not tell how many frames they get in return.
E.g. per tail call every unwinder is possible to return a different number of frames as the complexity and therefore the number of eBPF instructions vary between unwinders significantly. 
How should users set this value? What is the benefit of setting it to a lower one?
Can you provide numbers that show a noticeable impact on the CPU time or used memory, that supports making this a configurationable option?

Personally I would prefer if a limit should be set on the number of frames per stack trace, that this happens on the OTel collector side with a processor and not within eBPF.

Reply:
I would be happy to increase this limit unconditionally, rather than making it user-configurable. However, I was under the impression -- from the previous comment -- that we couldn't upstream a lower limit because it's necessary in order to allow the profiler to continue to work on old kernels.

If this isn't the case, and the maintainers are willing to accept a larger limit, I would be happy to replace this PR with one that does so. What do you think?